### DISCUSSION TITLE
I am confused, do I need coc.nvim for the latest stable neovim?

### DISCUSSION THREAD

**Discussion by WantToLearnJapanese** [neoclide/coc.nvim#3468] [Q&A] [answered]
As far as I know, we have LSP client on the latest neovim?

**Comment by oblitum** [comment, marked as answer]
Only if you want. NeoVim's embedded LSP isn't as mature.

**Reply by WantToLearnJapanese** [reply]
Ah, I wanted to ask the question if coc.nvim is LSP client or not. Can you please answer my question?

Thank you.

**Reply by oblitum** [reply]
Yes it's is, but it's not limited to that, as it has completion engine builtin (plus many other IDE-like utilitarians) and [an extension ecosystem of its own](https://github.com/neoclide/coc.nvim/wiki/Using-coc-extensions#why-are-coc-extensions-needed).
